drivers/SmartThings/zigbee-window-treatment/src/aqara/init.lua

```diff
@@ -0,0 +1,258 @@
+  elseif shadeLevel == 0 then
+    device:emit_event(capabilities.windowShade.windowShade.closed())
+  else
+    device:emit_event(capabilities.windowShade.windowShade.partially_open())
+  end
+end
+
+local function window_shade_level_cmd(driver, device, command)
+  local level = command.args.shadeLevel
+  device:send_to_component(command.component, WindowCovering.server.commands.GoToLiftPercentage(device, level))
+  device:emit_event(capabilities.windowShadeLevel.shadeLevel(level))
+end
+
+local function window_shade_open_cmd(driver, device, command)
+  local shadeLevel = device:get_field(SHADE_LEVEL) or 0
+  send_open_cmd(device, command.component)
+  emit_level_event(device, shadeLevel)
+end
+
+local function window_shade_close_cmd(driver, device, command)
+  local shadeLevel = device:get_field(SHADE_LEVEL) or 0
+  send_close_cmd(device, command.component)
+  emit_level_event(device, shadeLevel)
+end
+
+local function window_shade_pause_cmd(driver, device, command)
+  local shadeLevel = device:get_field(SHADE_LEVEL) or 0
+  device:send_to_component(command.component, WindowCovering.server.commands.Stop(device))
+  emit_level_event(device, shadeLevel)
+end
+
+local function motion_state_attr_handler(driver, device, value, zb_rx)
+  local state = value.value
+  device:set_field(SHADE_STATE, state)
+
+  if state == 0 then
+    local shadeLevel = device:get_field(SHADE_LEVEL) or 0
+    emit_level_event(device, shadeLevel)
+
+    local flag = device:get_field(INIT_STATE) or INIT_STATE_DONE
+    if flag == INIT_STATE_CLOSE then
+      device:set_field(INIT_STATE, INIT_STATE_REVERSE)
+      send_open_cmd(device, "main")
+    elseif flag == INIT_STATE_OPEN then
+      device:set_field(INIT_STATE, INIT_STATE_REVERSE)
+      send_close_cmd(device, "main")
+    elseif flag == INIT_STATE_REVERSE then
+      device:set_field(INIT_STATE, INIT_STATE_DONE)
+      read_pref_attribute(device)
+    end
+  elseif state == 1 then
+    device:emit_event(capabilities.windowShade.windowShade.opening())
+  elseif state == 2 then
+    device:emit_event(capabilities.windowShade.windowShade.closing())
+  end
+end
+
+local function current_position_attr_handler(driver, device, value, zb_rx)
+  local level = value.value
+  if level == 5.8774717541114e-39 then
+    level = 0
+  end
+  if level > 100 then
+    level = 100
+  end
+  level = utils.round(level)
+
+  device:set_field(SHADE_LEVEL, level)
+  device:emit_event(capabilities.windowShadeLevel.shadeLevel(level))
+
+  local shadeState = device:get_field(SHADE_STATE) or 0
+  if shadeState == 0 then
+    emit_level_event(device, level)
+  end
+end
+
+local function pref_attr_handler(driver, device, value, zb_rx)
+  local initialized = string.byte(value.value, 3) & 0xFF
+  local flag = device:get_field(INIT_STATE) or INIT_STATE_DONE
+  if flag == INIT_STATE_DONE then
+    device:emit_event(initialized  == 1 and initializedstate.initializedState.initialized() or initializedstate.initializedState.notInitialized())

```

Comment:
The only place that `supportedInitializedStates` is set it says the only supported value is `initialize` so it doesn't seem like emitting these values is supported.